web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js

```diff
@@ -0,0 +1,643 @@
+    setValue('columns', [...tableColumns, newColumn])
+  }
+
+  /**
+   * Remove a column at specified index
+   */
+  const removeColumn = index => {
+    let updatedColumns = [...tableColumns]
+    updatedColumns.splice(index, 1)
+    setTableColumns(updatedColumns)
+    setValue('columns', updatedColumns)
+  }
+
+  /**
+   * Add a new property field
+   * Checks for duplicate keys before adding
+   */
+  const addProperty = () => {
+    const hasDuplicateKeys = innerProps
+      .filter(item => item.key.trim() !== '')
+      .some(
+        (item, index, filteredItems) =>
+          filteredItems.findIndex(otherItem => otherItem !== item && otherItem.key.trim() === item.key.trim()) !== -1
+      )
+
+    if (hasDuplicateKeys) {
+      return
+    }
+
+    const newProperty = { key: '', value: '', required: false }
+
+    setInnerProps([...innerProps, newProperty])
+    setValue('propItems', [...innerProps, newProperty])
+  }
+
+  /**
+   * Remove a property field at specified index
+   */
+  const removeProperty = index => {
+    let updatedProps = [...innerProps]
+    updatedProps.splice(index, 1)
+    setInnerProps(updatedProps)
+    setValue('propItems', updatedProps)
+  }
+
+  /**
+   * Handle dialog close
+   * Resets form and clears state
+   */
+  const handleDialogClose = () => {
+    reset()
+    setInnerProps([])
+    setTableColumns([])
+    setValue('propItems', [])
+    setValue('columns', [])
+    setOpen(false)
+  }
+
+  /**
+   * Handle form submission
+   */
+  const handleFormSubmit = e => {
+    e.preventDefault()
+
+    return handleSubmit(submitForm(getValues()), handleValidationError)
+  }
+
+  /**
+   * Process form submission
+   * Validates data and dispatches create/update actions
+   */
+  const submitForm = formData => {
+    const hasDuplicateKeys = innerProps
+      .filter(item => item.key.trim() !== '')
+      .some(
+        (item, index, filteredItems) =>
+          filteredItems.findIndex(otherItem => otherItem !== item && otherItem.key.trim() === item.key.trim()) !== -1
+      )
+
+    const hasInvalidKeys = innerProps.some(prop => prop.invalid)
+
+    if (hasDuplicateKeys || hasInvalidKeys) {
+      return
+    }
+
+    trigger()
+
+    schema
+      .validate(formData)
+      .then(() => {
+        const properties = innerProps.reduce((acc, item) => {
+          acc[item.key] = item.value
+
+          return acc
+        }, {})
+
+        const tableData = {
+          name: formData.name,
+          comment: formData.comment,
+          columns: formData.columns,
+          properties
+        }
+
+        if (type === 'create') {
+          dispatch(createTable({ data: tableData, metalake, catalog, type: catalogType, schema: schemaName })).then(
+            res => {
+              if (!res.payload?.err) {
+                handleDialogClose()
+              }
+            }
+          )
+        } else {
+          const updates = genUpdates(initialTableData, tableData)
```

Comment:
the function `genUpdates` have no diff logic of table columns.

Reply:
Hi @LauraXia123 
I think I need some advice on handling column update behavior and GUI design. For example, if the original columns are:
- `first`, `long`, `nullable`
- `second`, `long`, `nullable`

And the new columns are:
- `second`, `long`, `nullable`

How can I tell whether the operation is a column name change or a column deletion?

I’ve thought of two solutions, both involving UI adjustments to avoid this issue:
1. Allow users to update one column at a time rather than making multiple modifications and then updating.
2. Modify the UI so that users can execute a sequence of commands and see the results as they expect (see example picture).

---

![image (1)](https://github.com/user-attachments/assets/7d409a26-4f5c-4935-b3cb-83df6981c604)

Reply:
Thank you for your reply. I will fix it within two days.